wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -284,15 +284,25 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
       (foundTxo.blockHash, blockHashOpt) match {
         case (None, Some(blockHash)) =>
           logger.debug(
-            s"Updating block_hash of txo=${transaction.txIdBE}, new block hash=${blockHash}")
-          val updateF =
-            spendingInfoDAO.update(
-              foundTxo
-                .copyWithBlockHash(blockHash = blockHash))
+            s"Updating block_hash of txo=${transaction.txIdBE}, new block hash=$blockHash")
+          // update block hash
+          val txoWithHash = foundTxo.copyWithBlockHash(blockHash = blockHash)
+
+          // If the utxo was marked reserved we want to update it to spent now
+          // since it has been included in a block
+          val unreservedTxo = txoWithHash.state match {
+            case TxoState.Reserved =>
+              txoWithHash.copyWithState(TxoState.PendingConfirmationsSpent)
+            case _: TxoState =>
```

Comment:
it's generally bad practice to do `_: TxoState` as if we add another type to `TxoState` we won't get a compiler warning telling us we need to update our code here. It's better to list ouf the possible states: 

```scala
case TxoState.ConfirmedSpent | TxoState.ConfirmedReceived | ... => 
```
So now if in the future we make a `ReservedColdStorage` (for example) we will get compiler warning telling us we need to update our code.